pkg/driver/controller.go

```diff
@@ -52,9 +52,17 @@ func (d *driver) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest)
 	}
 
 	volumePool := req.GetParameters()["volume_pool"]
+	var accessibleTopology []*csi.Topology
 	// if no volume_pool was provided try to use the topology information if provided
 	if volumePool == "" && req.GetAccessibilityRequirements() != nil {
 		volumePool = getAZFromTopology(req.GetAccessibilityRequirements())
+		accessibleTopology = []*csi.Topology{
+			{
+				Segments: map[string]string{topologyKey: volumePool},
+			},
+		}
+	} else if volumePool == "" && req.GetAccessibilityRequirements() == nil {
```

Comment:
Please use this as an `if` statement and return early the error.

Reply:
Done